### PR TITLE
Upgrade to Scala 2.12.11

### PR DESCRIPTION
**Pull request by leo-da**
Scala v2.12.6 has been around for longer than 2 years (released on Apr 27, 2018). I think it is time to upgrade, given:

https://github.com/scala/scala/releases/tag/v2.12.11
> Performance improvements in the collections library: algorithmic improvements and changes to avoid unnecessary allocations (list of PRs)
https://github.com/scala/scala/pulls?q=is%3Apr+milestone%3A2.12.11+is%3Aclosed+sort%3Acreated-desc+label%3Alibrary%3Acollections+label%3Aperformance

Sorry @remyhaemmerle-da but `/:`and `:\` have been deprecated in favor of `foldLeft` and `foldRight`.

Or should we upgrade straight to Scala 2.13.2? This may require more changes. We should do it in a separate PR.

### TODO
- [x] Run a perftest.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by leo-da** [comment]
## hotfix-scala-2.12.11-upgrade
```
$ bazel run daml-lf/scenario-interpreter:scenario-perf -- -f 4 -i 10 -wi 10
...
Result "com.daml.lf.speedy.perf.CollectAuthority.bench":
  133.682 ±(99.9%) 4.554 ms/op [Average]
  (min, avg, max) = (122.900, 133.682, 159.250), stdev = 8.095
  CI (99.9%): [129.128, 138.236] (assumes normal distribution)


# Run complete. Total time: 00:01:42

Benchmark                                                             (dar)             (scenario)  Mode  Cnt    Score   Error  Units
CollectAuthority.bench  //daml-lf/scenario-interpreter/CollectAuthority.dar  CollectAuthority:test  avgt   40  133.682 ± 4.554  ms/op
```

**Comment by leo-da** [comment]
### master, 1510d8ccd9707b027f8a8b78473355528e3ecc9b on my laptop
```
$ bazel run daml-lf/scenario-interpreter:scenario-perf -- -f 4 -i 10 -wi 10
...
Result "com.daml.lf.speedy.perf.CollectAuthority.bench":
  135.216 ±(99.9%) 4.420 ms/op [Average]
  (min, avg, max) = (117.186, 135.216, 147.701), stdev = 7.857
  CI (99.9%): [130.796, 139.636] (assumes normal distribution)


# Run complete. Total time: 00:01:41

Benchmark                                                             (dar)             (scenario)  Mode  Cnt    Score   Error  Units
CollectAuthority.bench  //daml-lf/scenario-interpreter/CollectAuthority.dar  CollectAuthority:test  avgt   40  135.216 ± 4.420  ms/op
```

**Comment by leo-da** [comment]
### Summary

We can safely merge this branch it does not hurt the performance.

File: daml-lf/interface/src/test/scala/com/digitalasset/daml/lf/iface/reader/InterfaceReaderSpec.scala [hunk 1]
```diff
@@ -164,7 +164,7 @@ class InterfaceReaderSpec extends WordSpec with Matchers with Inside {
     field -> Ast.TVar(var_)
 
   private def primField(field: Ref.Name, primType: Ast.BuiltinType, args: Ast.Type*) =
-    field -> ((Ast.TBuiltin(primType): Ast.Type) /: args)(Ast.TApp)
```
Comment (remyhaemmerle-da) [inline daml-lf/interface/src/test/scala/com/digitalasset/daml/lf/iface/reader/InterfaceReaderSpec.scala:167]: Thanks for taking care of that. 
I was on my personal backlog (probably to deep be ever reached)

**Comment by gerolf-da** [comment]
Thanks, @leo-da!

**Pull request merged**